posthog-android/src/main/java/com/posthog/android/replay/PostHogReplayIntegration.kt

```diff
@@ -453,13 +479,35 @@ public class PostHogReplayIntegration(
             events.add(it)
         }
 
-        if (events.isNotEmpty()) {
+        if (events.isNotEmpty() && sessionLongerThanMinDuration()) {
+            config.logger.log("Session replay events captured: " + events.size)
             events.capture()
+            events.clear()
         }
 
         status.lastSnapshot = wireframe
     }
 
+    private fun sessionLongerThanMinDuration(): Boolean {
+        //Check value only if threshold not crossed.
+        if (!minSessionThresholdCrossed) {
+            val serverMinDuration = replayPreferenceMap?.let { map ->
+                (map["minimumDurationMilliseconds"] as Number).toLong()
+            } ?: 0L
```

Comment:
Instead of reading from `replayPreferenceMap` (the disk), we can just update the `config.sessionReplayConfig.minSessionDurationMs` once, when the SDK is init, we do something similar with `snapshotEndpoint`

Reply:
I have made this change as per my understanding. 

On every launch `preloadSessionReplayFlag` is taking precedence over `loadFeatureFlags` hence the server value is taking 1 launch delta to be updated. 
Is this expected as well? This wasn't happening in the preference lookup. 

Do let me know if my understanding is incorrect here.

Reply:
Yes, it can happen that it will be using the last min duration and not the new one immediately but I will correct it as soon as the flags are loaded again, which is fine, its better than reading from the disk every time (too much IO), or awaiting a request before being able to use the feature, it's a trade-off we made.

Reply:
Understood. Will keep this change then.